### PR TITLE
On start, update git_pillar on second loop

### LINKED ISSUE #56356: Refresh git_pillar on master start
### Description of Issue

With PR #53621, it is possible to reduce the frequency of `git_pillar` updates.

However, `git_pillar` is not updated on server start, and one need to wait for `git_pillar_update_interval` to have the first run.

### Setup
Master config:
```yaml
git_pillar_update_interval: 604800 # 1 week
```

### Steps to Reproduce Issue
- Add a new repo in the git pillar configuration.
- Restart the master

Expected behavior:
- The new repo is available directly

Actual behavior:
- The new repo is available after 1 week :disappointed: 

### Versions Report
`v3000`

### PR DESCRIPTION
**Pull request by sathieu**
### What does this PR do?

Before this patch, the master would wait `git_pillar_update_interval` before doing the first update. When this setting is big (we set it to one week at work), this can lead to unfetched repositories.

### What issues does this PR fix or reference?

PR #53621

Fixes #56356.

### Previous Behavior
The first `git_pillar` update would run at _start time_ + `git_pillar_update_interval`.

### New Behavior
The first git_pillar update would run at _start time_.

### Tests written?

No

### Commits signed with GPG?

No

### CODE REVIEW

File: salt/master.py [hunk 1]
```diff
@@ -224,9 +224,10 @@ def run(self):
 
         # Make Start Times
         last = int(time.time())
-        last_git_pillar_update = last
-
+        # update git_pillar on next loop
         git_pillar_update_interval = self.opts.get('git_pillar_update_interval', 0)
+        last_git_pillar_update = last - git_pillar_update_interval + self.loop_interval
```
Comment (sathieu) [inline salt/master.py]: Alternatively, we can set `last_git_pillar_update = 0`, i.e. update `git_pillar` on first loop.

- Pros: code is simpler
- Cons: Higher load on start?

**Comment by Ch3LL** [comment]
can we get an associated issue open alongside this PR, so its clear what issue we are fixing and in case anyone else is running into the issue its easily discoverable?

**Comment by sathieu** [comment]
@Ch3LL Issue created: #56356.

**Review by Ch3LL — CHANGES_REQUESTED** [review]
any chance we can get a test case here to ensure it does not regress in the future?

**Comment by sathieu** [comment]
@Ch3LL I have no idea of how to test this.

**Comment by Ch3LL** [comment]
I would suggest a unit test here `tests/unit/test_master.py` and add a new test ckass for the maintenance class and add a test for the run function with `git_pillar_update_interval` set to different values.

**Comment by xeacott** [comment]
Hey @sathieu any follow up on this? I can help you write a test if you need for this PR. 😄

**Comment by sathieu** [comment]
@xeacott Yes, please help me with a test.

**Comment by sathieu** [comment]
> I would suggest a unit test here `tests/unit/test_master.py` and add a new test ckass for the maintenance class and add a test for the run function with `git_pillar_update_interval` set to different values.

@Ch3LL OK but how to escape the `while True` loop?

**Comment by xeacott** [comment]
I would suggest something like this to get you started...
```
class MaintenanceTestCase(TestCase, AdaptedConfigurationTestCaseMixin):
    """
    TestCase for salt.master.Maintenance class
    """

    def setUp(self):
        self.opts = salt.config.master_config(None)
        self.main_class = salt.master.Maintenance(self.opts)

    def tearDown(self):
        del self.main_class

    def test_run_func(self):
        """
        Test the run function inside Maintenance class.
        """
        with patch("salt.utils.process", autospec=True):
            with patch("salt.master", autospec=True):
                self.main_class.run()

```

**Comment by sathieu** [comment]
@Ch3LL Please review again. There are (passing :smile:) tests now!

**Comment by sathieu** [comment]
Also @xeacott please review ...

**Comment by xeacott** [comment]
Oh yes will do, thanks for writing those tests!

**Review by waynew — DISMISSED** [review]
@xeacott Tests look good to me - I'd double check the docs to make sure we don't specify a different behavior (I'm guessing not).

**Review by xeacott — DISMISSED** [review]

**Review by Ch3LL — DISMISSED** [review]

**Comment by sathieu** [comment]
@xeacott  You've marked this PR *Merge Ready*. I've now rebased and fixed `pre-commit` and `Pylint`. Hope you merge it now ...

**Review by Ch3LL — APPROVED** [review]

**Pull request merged**